modules/platform/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/coldstorage/MoveToColdStorage.java

```diff
@@ -50,8 +55,28 @@
     @Param(name = "save", required = false, values = "true")
     protected boolean save = true;
 
-    @OperationMethod(collector = DocumentModelCollector.class)
-    public DocumentModel run(DocumentModel doc) {
+    @OperationMethod
+    public DocumentModelList run(DocumentModelList documents) {
+        DocumentModelList result = new DocumentModelListImpl();
+        for (DocumentModel documentModel : documents) {
+            try {
+                result.add(moveContent(documentModel, false));
+            } catch (Exception e) {
+                log.error("Unable to move document: {} to cold storage", documentModel.getId(), e);
+            }
+        }
+        if (save) {
+            session.saveDocuments(result.toArray(new DocumentModel[0]));
+        }
+        return result;
```

Comment:
This behaves differently from `moveContent` when `save` is `true`. In `moveContent` you are returning the saved document while here the documents in `result` are not the ones returned by `CoreSession#saveDocument`.

Looking at the `saveDocuments` implementation, you could do your own iteration/stream :)

BTW, why not letting `moveContent` save the document? This would avoid multiple `if(save).`

Reply:
>This behaves differently from moveContent when save is true. In moveContent you are returning the saved document while here the documents in result are not the ones returned by CoreSession#saveDocument.
good point thanks

>Looking at the saveDocuments implementation, you could do your own iteration/stream :)
I agree and I thought my self same thing, but after a short reflection I told myself i prefers use the existing one to mainly if in future we re-write the CoreSession one to be for example more performante ... I will benefit

>BTW, why not letting moveContent save the document? This would avoid multiple if(save).
you mean save doc by doc, instead of save them ?